lib/widgets/sprite_widget.dart

```diff
@@ -45,20 +47,14 @@ class _SpritePainter extends CustomPainter {
   void paint(Canvas canvas, Size size) {
     final widthRate = size.width / _sprite.size.x;
     final heightRate = size.height / _sprite.size.y;
-
     final rate = min(widthRate, heightRate);
 
-    final w = _sprite.size.x * rate;
-    final h = _sprite.size.y * rate;
-
-    final double dx = _anchor.relativePosition.x * size.width;
-    final double dy = _anchor.relativePosition.y * size.height;
-
-    canvas.translate(
-      dx - w * _anchor.relativePosition.x,
-      dy - h * _anchor.relativePosition.y,
-    );
+    final paintSize = _sprite.size * rate;
+    final anchorPosition = _anchor.relativePosition;
+    final anchoredPosition = size.toVector2()..multiply(anchorPosition);
```

Comment:
Why not make size be a Vector2 already? So we can avoid this conversion?

Reply:
It is not under our control, it is the size in CustomPainter.

Reply:
Oh, didn't realized this was the widget class.